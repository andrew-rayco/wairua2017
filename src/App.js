import React, { Component } from 'react';
import './App.css';
import Feature from './Feature';
import Form from './Form';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Wairua Retreats</h1>
        <p>Holy shiz</p>
        <Feature />
        <Form />
      </div>
    );
  }
}

export default App;
