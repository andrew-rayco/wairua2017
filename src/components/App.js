import React, { Component } from 'react';
import { HashRouter as Router, Route } from 'react-router-dom'

import ScrollToTop from './ScrollToTop'

import Header from './Header'
import Home from './Home'
import Gallery from './Gallery'
import Workshops from './Workshops'
import PrivateClasses from './PrivateClasses'
import Create from './Create'
import Form from './Form'
import About from './About'
import Footer from './Footer'
import TeenCampSept from './TeenCampSept'


class App extends Component {
  render() {
    return (
      <Router>
        <ScrollToTop>
          <div className="App container">
            <Route path="/" component={Header} />
            <Route path="/" exact={true} component={Home} />
            <Route path="/gallery" exact={true} component={Gallery} />
            <Route path="/adventures" exact={true} component={Workshops} />
            <Route path="/adventures/private-classes" exact={true} component={PrivateClasses} />
            <Route path="/adventures/create-your-own" exact={true} component={Create} />
            <Route path="/form" component={Form} />
            <Route path="/about" component={About} />
            <Route path="/teen-camp-sept" component={TeenCampSept} />
            <Footer />
          </div>
        </ScrollToTop>
      </Router>
    );
  }
}

export default App;
