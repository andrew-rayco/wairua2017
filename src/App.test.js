import React from 'react';
import { shallow, mount } from 'enzyme'
import App from './App';

describe('<App/> component', () => {
  const app = shallow(<App />);

  it('renders without crashing', () => {
    expect(app.length).toEqual(1)
  });

  it('renders correctly', () => {
    expect(app).toMatchSnapshot()
  })

  it('renders a div with a class of container', () => {
    expect(app.find('.container').exists()).toBe(true)
  })

  it('renders a div with a class of App', () => {
    expect(app.find('.App').exists()).toBe(true)
  })

})

describe('<App/> component mounted has 7 children', () => {
  const app = mount(<App />)

  it('renders all seven children', () => {
    expect(app.find('.container').children().length).toEqual(7)
  })
})
