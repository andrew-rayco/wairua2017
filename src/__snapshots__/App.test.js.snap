// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App container"
>
        <Route
                component={[Function]}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/workshops"
        />
        <Route
                component={[Function]}
                path="/form"
        />
        <Route
                component={[Function]}
                path="/about"
        />
        <Route
                component={[Function]}
                path="/teen-camp-sept"
        />
        <Footer />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/workshops"
/>,
          <Route
            component={[Function]}
            path="/form"
/>,
          <Route
            component={[Function]}
            path="/about"
/>,
          <Route
            component={[Function]}
            path="/teen-camp-sept"
/>,
          <Footer />,
        ],
        "className": "App container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/workshops",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/form",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/teen-camp-sept",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App container"
>
          <Route
                    component={[Function]}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/workshops"
          />
          <Route
                    component={[Function]}
                    path="/form"
          />
          <Route
                    component={[Function]}
                    path="/about"
          />
          <Route
                    component={[Function]}
                    path="/teen-camp-sept"
          />
          <Footer />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/workshops"
/>,
            <Route
              component={[Function]}
              path="/form"
/>,
            <Route
              component={[Function]}
              path="/about"
/>,
            <Route
              component={[Function]}
              path="/teen-camp-sept"
/>,
            <Footer />,
          ],
          "className": "App container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/workshops",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/form",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/teen-camp-sept",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
